src/panels/config/devices/device-detail/ha-device-entities-card.ts

```diff
@@ -0,0 +1,175 @@
+  @property() public entities!: EntityRegistryEntry[];
+  @property() public narrow!: boolean;
+
+  private _entities = memoizeOne(
+    (
+      deviceId: string,
+      entities: EntityRegistryEntry[]
+    ): EntityRegistryEntry[] =>
+      entities
+        .filter((entity) => entity.device_id === deviceId)
+        .sort((ent1, ent2) =>
+          compare(
+            this._computeEntityName(ent1) || `zzz${ent1.entity_id}`,
+            this._computeEntityName(ent2) || `zzz${ent2.entity_id}`
+          )
+        )
+  );
+
+  protected render(): TemplateResult {
+    const entities = this._entities(this.deviceId, this.entities);
+    return html`
+      <ha-card>
+        ${entities.length
+          ? entities.map((entry: EntityRegistryEntry) => {
+              const stateObj = this.hass.states[entry.entity_id];
+              return html`
+                <paper-icon-item
+                  .entry=${entry}
+                  class=${classMap({ "disabled-entry": !!entry.disabled_by })}
+                >
+                  ${stateObj
+                    ? html`
+                        <state-badge
+                          .stateObj=${stateObj}
+                          slot="item-icon"
+                        ></state-badge>
+                      `
+                    : html`
+                        <ha-icon
+                          slot="item-icon"
+                          .icon=${domainIcon(computeDomain(entry.entity_id))}
+                        ></ha-icon>
+                      `}
+                  <paper-item-body two-line>
+                    <div class="name">${this._computeEntityName(entry)}</div>
+                    <div class="secondary entity-id">${entry.entity_id}</div>
+                  </paper-item-body>
+                  <div class="buttons">
+                    ${stateObj
+                      ? html`
+                          <paper-icon-button
+                            @click=${this._openMoreInfo}
+                            icon="hass:open-in-new"
+                          ></paper-icon-button>
+                        `
+                      : ""}
+                    <paper-icon-button
+                      @click=${this._openEditEntry}
+                      icon="hass:settings"
+                    ></paper-icon-button>
+                  </div>
+                </paper-icon-item>
+              `;
+            })
+          : html`
+              <div class="config-entry-row">
+                <paper-item-body two-line>
+                  <div>
+                    ${this.hass.localize(
+                      "ui.panel.config.devices.entities.none"
+                    )}
+                  </div>
+                </paper-item-body>
+              </div>
+            `}
+      </ha-card>
+    `;
+  }
+
+  private _openEditEntry(ev: MouseEvent): void {
+    const entry = (ev.currentTarget! as any).closest("paper-icon-item").entry;
+    showEntityRegistryDetailDialog(this, {
```

Comment:
It's weird that this is necessary. The update/confirm logic should happen in the dialog and we just should receive callbacks.